src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/address_union.rs

```diff
@@ -0,0 +1,449 @@
+use crate::prelude::*;
+
+#[macro_export]
+macro_rules! address_union {
+    (
+        $(
+            #[doc = $expr: expr]
+        )*
+        $union_name: ident,
+        $(
+            $variant_name: ident,
+            $variant_type: ty
+        )+
+    ) => {
+        paste! {
```

Comment:
`paste!` is a macro which allows for `[< word1 _ word2 _ $use_variable _ word4 >]` syntax, so that we can create identifiers (type names, method names, function names etc) using the variables (`$union_name`, or `$variant_type` or `$variant_type`).